docs/adr/adr_improved_frontend_dx.md

```diff
@@ -0,0 +1,58 @@
+# ADR: Improved Frontend Developer Experience
+Decision Deadline: As soon as possible
+
+Author: [Felipe](https://github.com/fhenrich33)
+
+## Status
+
+Proposed, Implementing
+
+## Context
+Frontend development experience in its current form is quite lacking. It imposes a lot of artificial difficulties, tooling doesn't work as expected and adds friction to our FE contributors.
+
+Local installation of packages and tooling might not play well with the codebase. Things like TS, ESlint, and Prettier might not work at all without a lot of fiddling. 
+
+We want to make FE development better.
+
+## Decision drivers
+
+1. We must make FE onboarding and development seamless.
+2. Implementation must play well with the current FE setup (React/Vite).
+3. To add to the prior point: it must be maintainable with minimal hassle in the long term.
+4. Implementation must work well with our CI/CD pipeline.
+5. We must NOT interfere with BE tooling and development.
```

Comment:
There's one overlap of BE and FE in terms of tools: the GraphQL schema is currently being defined in `.graphql` files [here](https://github.com/boxwise/boxtribute/tree/master/back/boxtribute_server/graph_ql/definitions). We use the `graphql-codegen` tool to create TS type definitions from these for the FE.

As a BE dev I update/extend the schema files and want the TS type definitions to be aligned. If FE tooling setup is simple I would install it (if not, the current situation becomes reversed, slightly). Alternatively we could implement a bot that operates on PRs and applies potential TS type changes back to the current PR branch. What do you think is a good middle-ground solution?

Reply:
I think that is a reasonable solution to try out in the MVP build @fhenrich33 what do you think?